src/components/tabular-grid-render/tests/tabular-grid-render.e2e.ts

```diff
@@ -0,0 +1,21 @@
+import { E2EElement, E2EPage, newE2EPage } from "@stencil/core/testing";
+import { basicModel } from "../../../showcase/assets/components/tabular-grid-render/models";
+
+describe("[ch-tabular-grid-render]", () => {
+  let page: E2EPage;
+  let tabularGridRef: E2EElement;
+
+  beforeEach(async () => {
+    page = await newE2EPage({
+      html: `<ch-tabular-grid-render></ch-tabular-grid-render>`,
+      failOnConsoleError: true
+    });
+    tabularGridRef = await page.find("ch-tabular-grid-render");
+    tabularGridRef.setProperty("model", basicModel);
+    await page.waitForChanges();
+  });
+
+  it("should have Shadow DOM", async () => {
+    expect(tabularGridRef.shadowRoot).toBeDefined();
```

Comment:
I need to update some unit tests, because in some cases we use `toBeDefined()` which is wrong, because it strictly checks agains `undefined`, but the `shadowRoot` is `null` when the control does not support Shadow DOM.

```suggestion
    expect(tabularGridRef.shadowRoot).toBeTruthy();
```